CoverArtUtils: Fix Reload From File/Folder, Updates Wrong Cover Art

fatihemreyildiz opened the pull request:
This PR aims to solve this bug mentioned on [Launchpad](https://bugs.launchpad.net/mixxx/+bug/1985843).

This simply fixes to wrong cover art update if the names has more than one "." dot. I
track.1.mp3
track.2.mp3
track.3.mp3
track.4.jpg
track.5.jpg
 
If track.1-track.2-track.3's cover arts updated via "Reload from file/folder" it updates it with track.4.jpg. 

With this change, it looks for the complete full name, so track.1.mp3 won't be updated with track.4.jpg. 

But with that change, this can change the old behavior, if the files named as;
track.1.mp3.jpg
track.1.mp3
updating wouldn't not update the cover art, to not change the behavior of old updating, there is an another block added. If complete full name is not working, this will look for file name of the track `"track.1.mp3"` and complete name of the cover art `"track.1.mp3".jpg`

fatihemreyildiz committed:
cautils&fileinfo: compBaseName add for cart update

daschuer commented:
Thank you. This looks like a bug fix for 2.3.

fatihemreyildiz commented:
Yes, that is happening in 2.3 :+1:

daschuer commented on src/library/coverartutils.cpp:
```diff
@@ -116,13 +116,17 @@ CoverInfoRelative CoverArtUtils::selectCoverArtForTrack(
+            } else if (bestType > TRACK_BASENAME &&
+                    coverBaseName.compare(trackFile.fileName(),
+                            Qt::CaseInsensitive) == 0) {
+                bestInfo = &file;
```
```suggestion
                bestType = TRACK_BASENAME;
                bestInfo = &file;
```

fatihemreyildiz replied:
I've added the missing line.

daschuer reviewed (changes requested):
One missing line.

Can you rebase this on 2.3 than. Thank you.

fatihemreyildiz committed:
coverartutils: bestType added.

daschuer reviewed (approved):
LGTM

daschuer commented:
Does anyone else want to have a look?

daschuer commented:
Thank you!

daschuer commented:
Ups, this is now broken in the 2.3 branch: https://github.com/mixxxdj/mixxx/runs/7979410003
I did not realize that the CI was still testing the main merge after changing the merge target. 
I will push a fix directly to 2.3

daschuer commented:
https://github.com/mixxxdj/mixxx/commit/517d7a64a51d439be309f6fcbcd73cce4820075e

fatihemreyildiz commented:
Thank you for taking care. 

I guess the file `trackfile.h` in 2.3 branch, changed to `fileinfo.h` in main branch right? I was trying to fix it and added the completeBaseName in `file.h` in main branch and while re-basing to 2.3 there was no `fileinfo.h` so I deleted completeBaseName, and compiled successfully. 

I just didn't get it why it failed. I hope I didn't do something missing-wrong.
